hal/halx86/xbox/part_xbox.c

```diff
@@ -118,20 +119,38 @@ HalpXboxDeviceHasXboxPartitioning(IN PDEVICE_OBJECT DeviceObject,
         return STATUS_NO_MEMORY;
     }
 
+    Offset.QuadPart = 0;
+    Status = HalpXboxReadSector(DeviceObject, SectorSize, &Offset, SectorData);
+    if (!NT_SUCCESS(Status))
+    {
+        goto Cleanup;
+    }
+
+    HasMBRPartitioning = (*((UCHAR *) SectorData + SectorSize - 2) == 0x55 && *((UCHAR *) SectorData + SectorSize - 1) == 0xAA);
+    if (HasMBRPartitioning)
+    {
+        *HasXboxPartitioning = FALSE;
+        goto Cleanup;
+    }
+
     Offset.QuadPart = XBOX_SIGNATURE_SECTOR * SectorSize;
     Status = HalpXboxReadSector(DeviceObject, SectorSize, &Offset, SectorData);
     if (!NT_SUCCESS(Status))
     {
-        return Status;
+        goto Cleanup;
     }
 
     DPRINT("Signature 0x%02x 0x%02x 0x%02x 0x%02x\n",
            *((UCHAR *) SectorData), *((UCHAR *) SectorData + 1), *((UCHAR *) SectorData + 2), *((UCHAR *) SectorData + 3));
     *HasXboxPartitioning = (XBOX_SIGNATURE == *((ULONG *) SectorData));
+Cleanup:
     ExFreePool(SectorData);
```

Comment:
(Note: maybe it would be great to use ExAllocatePoolWithTag / ExFreePoolWithTag, as done in HalpXboxIoReadPartitionTable() ?)

Reply:
Improved :)